lib/IAMClient.js

```diff
@@ -482,6 +336,34 @@ class IAMClient {
         req.end();
     }
 
+    getObj(ret, cb) {
+        let obj = {};
+        let err = null;
+        if (ret.length === 0) {
+            return cb(null, obj);
+        }
+        if (ret[0] !== '{') {
+            return parseString(ret, {
+                explicitArray: false,
+            }, (err, result) => {
+                if (err) {
+                    return cb(err);
+                }
+                return cb(null, result);
+            });
+        }
+        try {
```

Comment:
Can you move this try catch into it's own function? Also, you may not want to attach `ret` to `err` object as this will be logged.